extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/internal/json/WeatherUndergroundJsonUtils.java

```diff
@@ -112,7 +113,8 @@ public static Integer convertToInteger(String value) {
      */
     public static BigDecimal convertToBigDecimal(String value) {
         BigDecimal result = null;
-        if (value != null && !value.isEmpty() && !value.equalsIgnoreCase("N/A") && !value.equalsIgnoreCase("NA")) {
+        if (value != null && !value.isEmpty() && !value.equalsIgnoreCase("N/A") && !value.equalsIgnoreCase("NA")
+                && !value.equals("-") && !value.equals("--")) {
```

Comment:
Please change this to if (!(value == null || value.isEmpty() || value.equalsIgnoreCase(....)), this makes it better to read.